### PR TITLE
Add Rspec and gems for test

### PR DESCRIPTION
**Pull request by diliaaragon**
- Add Rspec at project

- Add Database_cleaner gem and config

- Add Shoulda-Matchers gem and its configurations

- Add Factory_bot_rails gem and config

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -33,5 +33,6 @@ class Application < Rails::Application
     # Don't generate system test files.
     config.generators.system_tests = nil
     config.active_storage.variant_processor = :vips
+    config.factory_bot.definition_file_paths = ["custom/factories"]
```
Comment (juankhazbun) [inline config/application.rb]: No tenias que quitar esa linea?

File: spec/factories/user.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :user do
+    first_name { "John" }
```
Comment (juankhazbun) [inline spec/factories/user.rb]: Utiliza Faker

**Review by juankhazbun — CHANGES_REQUESTED** [review]

File: spec/factories/product.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :product do
+    name { "product" }
+    description  { "un producto" }
```
Comment (juankhazbun) [inline spec/factories/product.rb]: Utiliza Faker

**Review by juankhazbun — CHANGES_REQUESTED** [review]

File: config/application.rb (hunk 1 above)
Comment (diliaaragon) [reply config/application.rb]: listo

File: spec/factories/product.rb (hunk 3 above)
Comment (diliaaragon) [reply spec/factories/product.rb]: listo

File: spec/factories/user.rb (hunk 2 above)
Comment (diliaaragon) [reply spec/factories/user.rb]: listo

**Review by juankhazbun — APPROVED** [review]

**Pull request merged**